core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -225,6 +220,8 @@ class Partition(val topic: String,
       if (isNewLeader) {
         // construct the high watermark metadata for the new leader replica
         leaderReplica.convertHWToLocalOffsetMetadata()
+        // mark local replica as the leader
+        leaderReplicaIdOpt = Some(localBrokerId)
```

Comment:
Maybe the comment could be `// mark local replica as the leader after converting hw` so that it's clear that the order is important here.